app/interfaces/api/entities/ccr/advocate_claim.rb

```diff
@@ -68,26 +68,22 @@ def adapted_advocate_category
           ::CCR::AdvocateCategoryAdapter.code_for(object.advocate_category) if object.advocate_category.present?
         end
 
-        def basic_fee_adapter
-          ::CCR::Fee::BasicFeeAdapter.new
+        def adapted_basic_fee
+          ::CCR::Fee::BasicFeeAdapter.new(object)
         end
 
         def basic_fees
-          fee = basic_fee_adapter.call(object)
-          [].tap do |arr|
-            arr << fee if fee.claimed?
-          end
+          fee = adapted_basic_fee
+          [].tap { |arr| arr << fee if fee.claimed? }
         end
 
-        def fixed_fee_adapter
-          ::CCR::Fee::FixedFeeAdapter.new
+        def adapted_fixed_fee
+          ::CCR::Fee::FixedFeeAdapter.new.call(object)
         end
 
         def fixed_fees
-          fee = fixed_fee_adapter.call(object)
-          [].tap do |arr|
-            arr << fee if fee.claimed?
-          end
+          fee = adapted_fixed_fee
+          [].tap { |arr| arr << fee if fee.claimed? }
```

Comment:
**[nit pick/not a blocker]** Seems there's no need to assign `fee` as long as  `adapted_fixed_fee` is memoized:

```ruby
adapted_fixed_fee.claimed? ? [adapted_fixed_fee] : []
```

Reply:
yeah, have been wanting to refactor this